app/controllers/apis/roles_controller.rb

```diff
@@ -0,0 +1,52 @@
+module Apis
+  class RolesController < ApiController
+    def add_roles
+      response = ExistingRole.create(name: params[:name]) # if request_is_valid_post?
+      if response.errors.any?
+        render_403(response)
+      else
+        render json: {
```

Comment:
It would be better if you can align the response format with RESTful specification. e.g creating an entity would return the new entity itself as the response body. I think there would be some rails gem to achieve it.